include/revng/ABI/CallingConventionTrait.h

```diff
@@ -0,0 +1,1083 @@
+#pragma once
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include "revng/Model/Binary.h"
+#include "revng/Model/Register.h"
+#include "revng/Model/Type.h"
+
+namespace abi {
+
+/// A trait struct used for describing the specifics of the calling convention
+/// for an ABI relevant for revng model.
+///
+/// To register a new ABI for use with the `abi::ABI` template, add
+/// a specialization of this trait.
+template<model::abi::Values V>
+struct CallingConventionTrait {
+
+  /// States whether the ABI is focused on preserving the argument order.
+  ///
+  /// Here's an example function:
+  /// ```
+  /// struct Big; // Is really big, so it can only be passed in memory.
+  /// void function(Big, signed, float, unsigned);
+  /// ```
+  ///
+  /// If the ABI is position based, the arguments would be passed in
+  /// - memory for the `Big` argument (and a pointer to it in the first GRP).
```

Comment:
```suggestion
  /// - memory for the `Big` argument (and a pointer to it in the first GPR).
```

There are multiple instances of this, please `sed` them out.

Reply:
[Done](https://github.com/revng/revng/pull/98/commits/a76430867040b4d6bef2f75568023a7f1622007e).